deploy/cmd/create.go

```
@@ -0,0 +1,171 @@
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			log := logger.With("cli", "create")
+			dd, err := NewDeploymentData(args[0], cmd.Flags(), clientCtx)
+			if err != nil {
+				return err
+			}
+
+			ctx, cancel := context.WithCancel(context.Background())
+			group, _ := errgroup.WithContext(ctx)
+
+			// Listen to on chain events and send the manifest when required
+			group.Go(func() error {
+				if err = ChainEmitter(ctx, clientCtx, DeploymentDataUpdateHandler(dd), SendManifestHander(clientCtx, dd)); err != nil {
+					log.Error("error watching events", err)
+				}
+				return err
+			})
+
+			// Send the deployment creation transaction
+			group.Go(func() error {
+				if err = TxCreateDeployment(clientCtx, cmd.Flags(), dd); err != nil {
+					log.Error("error creating deployment", err)
+				}
+				return err
+			})
+
+			// Wait for the leases to be created and then start polling the provider for service availability
+			group.Go(func() error {
+				if err = WaitForLeasesAndPollService(clientCtx, dd, cancel); err != nil {
+					log.Error("error listening for service", err)
+				}
+				return err
+			})
+
+			return group.Wait()
+		},
+	}
+
+	cmd.Flags().String(flags.FlagChainID, "", "The network chain ID")
+	if err := viper.BindPFlag(flags.FlagChainID, cmd.Flags().Lookup(flags.FlagChainID)); err != nil {
+		return nil
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+	dcli.AddDeploymentIDFlags(cmd.Flags())
+
+	return cmd
+}
+
+// WaitForLeasesAndPollService waits for leases
+func WaitForLeasesAndPollService(clientCtx client.Context, dd *DeploymentData, cancel context.CancelFunc) error {
+	log := logger
+	pclient := pmodule.AppModuleBasic{}.GetQueryClient(clientCtx)
+	timeout := time.After(150 * time.Second)
+	tick := time.Tick(500 * time.Millisecond)
+	for {
+		select {
+		case <-timeout:
+			log.Info("timed out (90s) listening for deployment to be available")
+			cancel()
+			return nil
```

I think we need to return an error here so that `group.Wait()` returns the same message and indicates the command failed.